### PR TITLE
Improve documentation of utilities in docs/FEATURES.md

### PR DESCRIPTION
**Pull request by aprotyas**
This PR does the following:

- Improve Doxygen linking between entities listed in `docs/FEATURES.md` and actual class/function documentation.
- Adds more detail about specific header files: particularly `find_library.hpp`, string helpers, and `shared_library.hpp`.
- Adds undocumented utilities into `docs/FEATURES.md`, namely: `process.hpp`, `get_env.hpp`, and `scope_exit.hpp`.
- Fix some grammatical nitpicks.

Signed-off-by: Abrar Rahman Protyasha <abrar@openrobotics.org>

### CODE REVIEW

File: docs/FEATURES.md [hunk 1]
```diff
@@ -77,5 +79,58 @@ These headers include simple functions for joining a container into a single str
...
+// Obtain platform-specific library name from the provided base name.
+const std::string library_name = rcpputils::get_platform_library_name("foo_lib");
+
+// The shared library is loaded in the constructor.
+auto library = std::make_shared<rcpputils::SharedLibrary>(library_name);
+
+// Check if a symbol exists in the shared library.
+// If so, obtain a pointer to the symbol.
+if(library -> has_symbol("example_symbol")) {
+    std::shared_ptr<void> symbol(library -> get_symbol("example_symbol"));
```
Comment (clalancette) [inline docs/FEATURES.md:96]:
```suggestion
    std::shared_ptr<void> symbol(library->get_symbol("example_symbol"));
```

File: docs/FEATURES.md [hunk 2]
```diff
@@ -77,5 +79,58 @@ These headers include simple functions for joining a container into a single str
...
+    // Use shared library symbol pointer.
+}
+```
+
+## Process helpers {#process-helpers}
+The `rcpputils/process.hpp` header contains process utilities.
+
+Namely, this header provides the `rcpputils::get_executable_name()` function, which retrieves and returns the current program name as a string. It is a *thread-safe* function.
```
Comment (clalancette) [inline docs/FEATURES.md:104]: I'm not sure I would say it is "thread-safe" here.  While that is true, this is duplicating the API documentation.

**Review by clalancette — CHANGES_REQUESTED** [review]
I've left a few things to change in here.

File: docs/FEATURES.md (hunk 2 above)
Comment (aprotyas) [reply docs/FEATURES.md]: True. I wasn't sure if it was particularly important for users to know. Removed in a908398.

File: docs/FEATURES.md [hunk 3]
```diff
@@ -41,32 +44,31 @@ ResultType some_function(ArgType arg)
...
+  * For dynamically loading user-defined plugins in C++, please use [`pluginlib`](https://github.com/ros/pluginlib) instead.
 
 ## String Helpers {#string-helpers}
-In `rcpputils/join.hpp` and `rcpputils/split.hpp`
+String helper utilities can be found in the `rcpputils/find_and_replace.hpp`, `rcpputils/join.hpp`, and `rcpputils/split.hpp` headers.
+
+These headers include simple functions that facilitate find and replace functionality for strings, joining a container into a single string, and splitting a string into a container of values.
 
-These headers include simple functions for joining a container into a single string and splitting a string into a container of values.
+The functions of interest are namesakes of each of the specified header files.
```
[the remark this answers is not in the document]
Comment (aprotyas) [reply docs/FEATURES.md]: Addressed in a908398.

File: docs/FEATURES.md [hunk 4]
```diff
@@ -41,32 +44,31 @@ ResultType some_function(ArgType arg)
...
+See [cppreference](https://en.cppreference.com/w/cpp/types/endian) for more information.
 
 ## Library Discovery {#library-discovery}
+The `rcpputils/find_library.hpp` facilitates finding a library located in the OS's library paths environment variable.
 
-In `rcpputils/find_library.hpp`:
-
-*   `find_library(library_name)`: Namely used for dynamically loading RMW
+* `rcpputils::find_library_path(const std::string &)`: Namely used for dynamically loading RMW
     implementations.
```
[the remark this answers is not in the document]
Comment (aprotyas) [reply docs/FEATURES.md]: Addressed in a908398.

File: docs/FEATURES.md [hunk 5]
```diff
@@ -77,5 +81,58 @@ These headers include simple functions for joining a container into a single str
...
+```c++
+// Obtain platform-specific library name from the provided base name.
+const std::string library_name = rcpputils::get_platform_library_name("foo_lib");
+
+// The shared library is loaded in the constructor.
+auto library = std::make_shared<rcpputils::SharedLibrary>(library_name);
+
+// Check if a symbol exists in the shared library.
+// If so, obtain a pointer to the symbol.
+if(library -> has_symbol("example_symbol")) {
```
Comment (clalancette) [inline docs/FEATURES.md:97]:
```suggestion
if (library->has_symbol("example_symbol")) {
```

**Review by clalancette — APPROVED** [review]
Whoops, I missed one minor thing inline.  I'm going to approve anyway; feel free to just fix that before you merge.

**Comment by ros-discourse** [comment]
This pull request has been mentioned on **ROS Discourse**. There might be relevant details there:

https://discourse.ros.org/t/ros-2-tsc-meeting-minutes-2021-9-16/22372/1

**Pull request merged**